ros2_generate_interface_docs/ros2_generate_interface_docs/utils.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from io import StringIO
+import os
+import sys
+
+import em
+
+from rosidl_parser.definition import AbstractNestedType, BASIC_TYPES, NamespacedType
+
+_TEMPLATES_DIR = 'templates'
+IGNORED_KEYS = ['__slots__', '__doc__', '_fields_and_field_types', 'SLOT_TYPES', '__init__',
+                '__repr__', '__eq__', '__hash__', 'get_fields_and_field_types', '__module__']
+
+
+def generate_raw_text(raw_text):
+    raw_test_str = ''
+    for line in raw_text.split(os.linesep):
+        line = line.replace(' ', '&nbsp;')
+        parts = line.split('#')
+        if len(parts) > 1:
+            raw_test_str = raw_test_str + parts[0] \
+                + '<font color="blue">#%s</font><br/>' % ('#'.join(parts[1:]))
+        else:
+            raw_test_str = raw_test_str + '%s<br/>' % parts[0]
+    return raw_test_str
+
+
+def resource_name(resource):
+    """Return the resource name."""
+    if '/' not in resource:
+        return '', '', resource
+    values = resource.split('/')
+    if len(values) != 3:
+        raise ValueError(...)
+    return tuple(values)
+
+
+def get_templates_dir():
+    """Return template directory."""
+    return os.path.join(os.path.dirname(__file__), _TEMPLATES_DIR)
+
+
+def copy_css_style(folder_name):
+    """Copy the css style file in the folder_name."""
+    style_css, _ = load_template('styles.css')
+    with open(os.path.join(folder_name, 'styles.css'), 'w') as f:
+        f.write(style_css)
+
+    msg_style_css, _ = load_template('msg-styles.css')
+    with open(os.path.join(folder_name, 'msg-styles.css'), 'w') as f:
+        f.write(msg_style_css)
+
+
+def load_template(filename):
+    """
+    Look up file within rosdoc ROS package.
+
+    return its content, may sys.exit on error.
+    Contents are cached with filename as key.
+
+    :returns: cached file contents
+    """
+    filename = os.path.join(get_templates_dir(), filename)
+    if not os.path.isfile(filename):
+        sys.stderr.write("Cannot locate template file '%s'\n" % (filename))
```

Comment:
This file path should be checked in main after parsing arguments. If this file doesn't exist here I think it should raise an exception instead of exiting

Reply:
0cdaabc522a58ad82a21dbc6c7cddc894b92766b